### ISSUE TITLE
Proxy setting gets removed when installing packages

### ISSUE THREAD

**Issue by hfjallemark** [halfzebra/create-elm-app#160]
Every time I install a new package the `proxy` setting in `elm-package.json` gets removed. Am I doing something wrong or is this how elm package works? If so, should proxy be moved out of elm-package.json?

**Comment by halfzebra** [comment]
@hfjallemark thanks for reporting this issue! It looks like this behavior originates in [elm-package](https://github.com/elm-lang/elm-package), which rewrites `elm-package.json` on every `elm-package install`

I can't think of a good solution for now.

@fobos any suggestions?

**Comment by gyzerok** [comment]
Might be worth having create-elm-app.json?

**Comment by fobos** [comment]
Now I have the same idea like @gyzerok

**Comment by halfzebra** [comment]
@gyzerok @fobos what if we'll add `package.json` to the template instead of introducing custom config file?

**Comment by gyzerok** [comment]
It seems a bit misleading to me. Will there be any "packages"? How is it related to npm? How would it's content look like?

**Comment by hfjallemark** [comment]
I agree, if there are no other dependencies to npm I think it could be a bit confusing to have a package.json file there just for a setting. Not that I have a good alternative :-)

**Comment by hfjallemark** [comment]
One alternative that doesn’t require any new config files could be to pass it in as a command line argument (similar to `ELM_DEBUGGER`), e.g: `ELM_PROXY=http://localhost:3000 elm-app start`. Since I already have my start script in `package.json` setup it would just mean adding one more argument.

**Comment by jhillyerd** [comment]
I also got bit by this, and it took my quite a while to figure out what was going on due to being new to Elm.  I think having a separate config would be nice.

If a config file or env is added, I would appreciate the ability to configure a [websocket proxy](https://stackoverflow.com/questions/31654239/proxy-websockets-connection-within-webpack-dev-server/39233132) as I believe I will have to eject from CEA to continue my current project.

**Comment by halfzebra** [comment]
@hfjallemark this sounds like a pretty good option.

@jhillyerd sorry to hear that you've had to eject. I'll see if there's a better way.

@gyzerok I don't think I'll have time to work on the "elm-app config file" idea before the next Elm version is out. What do you think about the env variable?

**Comment by jhillyerd** [comment]
Turns out (after ejecting and reading some docs) that I don't need to eject.  I filed #251 to recommend an update to docs.  Please be sure to preserve this (proxy as object) functionality in whatever replaces elm-package.json.  Thanks!

**Comment by gyzerok** [comment]
@halfzebra the idea of passing env variable seems to be nice. However before starting with it I would suggest to figure out how feasible it is.

In my current project configuration looks like this:

```
  "proxy": {
    "/api": {
      "target": "https://backend.com",
      "secure": false
    }
  },
```

How would specify the same with env variable?

It'd be nice to have a bit more use-cases from different people. I am mostly worried how feasible it is to do non-trivial configuration using env var.

**Comment by akavel** [comment]
How about adding some wrapper before and after the `elm-package` call, which would preserve selected sub-keys of the JSON? For example by extracting them to a temporary JSON file, then running `elm-package`, then merging the temporary JSON back into `elm-package.json`?

I'm not sure how `elm-app` works internally, so no idea if this is feasible...
